test/test_pull1663.m

```diff
@@ -0,0 +1,185 @@
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% 1. create input data
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+%% create mesh
+
+% from test_pull1427.m
+% dull segmentation
+segprob = [];
+segprob.brain = false(10,10,10); segprob.brain(4:7,4:7,4:7) = true;
+segprob.skull = false(10,10,10); segprob.skull(3:8,3:8,3:8) = true;
+segprob.scalp = false(10,10,10); segprob.scalp(2:9,2:9,2:9) = true;
+% segprob.air = true(10,10,10);
+% segprob.air(2:9,2:9,2:9) = false;
+%segprob.brain = false(11,11,11); segprob.brain(4:8,4:8,4:8) = true;
+%segprob.skull = false(11,11,11); segprob.skull(3:9,3:9,3:9) = true;
+%segprob.scalp = false(11,11,11); segprob.scalp(2:10,2:10,2:10) = true;
+segprob.dim = size(segprob.brain);
+segprob.unit = 'cm';
+segprob.coordsys = 'ctf';
+segprob.transform = eye(4);
+segprob.transform(1,4) = -0.5;
+segprob.transform(2,4) = -0.5;
+segprob.transform(3,4) = -0.5;
+
+% visualize the segmentation 
+% it is more difficult to visualize a probabilistic segmentation than an indexed one
+segindx = ft_datatype_segmentation(segprob, 'segmentationstyle', 'indexed');
+
+cfg = [];
+cfg.funparameter = 'seg';
+cfg.method = 'ortho';
+cfg.location = [5 5 5]; % this is the center of the volume, in this plot it will be rounded off to the nearest voxel
+ft_sourceplot(cfg, segindx);
+
+% hexa mesh
+cfg = [];
+cfg.shift = 0.3;
+cfg.method = 'hexahedral';
+mesh_vol_hex = ft_prepare_mesh(cfg, segprob);
+
+figure
+ft_plot_ortho(segindx.seg, 'transform', segindx.transform, 'location', [5 5 5], 'style', 'intersect');
+hold on
+ft_plot_mesh(mesh_vol_hex, 'surfaceonly', false, 'facecolor', 'none', 'edgecolor', 'm');
+view(120, 30)
+
+
+% tetra mesh
+cfg = [];
+cfg.method = 'tetrahedral';
+mesh_vol_tet = ft_prepare_mesh(cfg, segprob);
+
+
+figure
+ft_plot_ortho(segindx.seg, 'transform', segindx.transform, 'location', [5 5 5], 'style', 'intersect');
+hold on
+ft_plot_mesh(mesh_vol_tet, 'surfaceonly', false, 'facecolor', 'none', 'edgecolor', 'm');
+view(120, 30)
+
+%% define sensors
+
+% i manually passed 5 coils and fixed projections 
+% or maybe take some from a ctf file? something like this (from test_pull1377.m)
+
+% for MEG data + sensor info
+% 
+% load(dccnpath('/home/common/matlab/fieldtrip/data/test/latest/raw/meg/preproc_ctf151.mat'), 'data');
+% datameg = data;
+% clear data
+
+coils = [5 5 12; 5 12 5; -2 5 5; 12 5 5; 5 -2 5];
+projections = [0 0 1 ; 0 1 0; -1 0 0; 1 0 0; 0 -1 0 ];
+
+figure
+ft_plot_ortho(segindx.seg, 'transform', segindx.transform, 'location', [5 5 5], 'style', 'intersect');
+hold on
+quiver3(coils(:,1),coils(:,2),coils(:,3),projections(:,1),projections(:,2),projections(:,3),'bo')
+
+meg_labels = cellstr(strings(1,size(coils,1)));
+for i=1:size(coils,1)
+  meg_labels(i) = cellstr(strcat('meg',num2str(i)));
+end
+
+sens = [];
+sens.coilpos = coils;
+sens.coilori = projections;
+sens.chanpos = coils;
+sens.chanori = projections;
+sens.label = meg_labels;
```

Comment:
please add `sens.tra=eye(5)` to make it explicit that there mapping from coils to channels is one-to-one (i.e. magnetometers).

Reply:
added in commit 4375db7